build: make build error on fresh clone

First `make build` on a fresh clone of `cockroach` fails. Something tries to use the `errors` package before the `vendor` repo is cloned. Not a big deal, it works if you then run it again. But it might be dismaying to an external contributor that's trying to get started. CC @knz

```
$ make build
GOPATH set to /go2
Cleaning old generated files.
# Special case for 'prefix' variable, which is required by Homebrew.
Running make with -j8
GOPATH set to /go2
Cleaning old generated files.
Detected change in build system. Rebooting Make.
# Special case for 'prefix' variable, which is required by Homebrew.
Running make with -j8
GOPATH set to /go2
Cleaning old generated files.
installing githooks/post-rewrite
installing githooks/commit-msg
git submodule update --init --recursive
installing githooks/post-merge
installing githooks/prepare-commit-msg
installing githooks/post-checkout
go install -v ./pkg/cmd/prereqs
mkdir -p pkg/sql/parser/gen
set -euo pipefail; \
awk '/func.*sqlSymUnion/ {print $(NF - 1)}' pkg/sql/parser/sql.y | \
sed -e 's/[]\/$*.^|[]/\\&/g' | \
sed -e "s/^/s_(type|token) <(/" | \
awk '{print $0")>_\\1 <union> /* <\\2> */_"}' > pkg/sql/parser/gen/types_regex.tmp; \
sed -E -f pkg/sql/parser/gen/types_regex.tmp < pkg/sql/parser/sql.y | \
awk -f pkg/sql/parser/replace_help_rules.awk | \
sed -Ee 's,//.*$,,g;s,/[*]([^*]|[*][^/])*[*]/, ,g;s/ +$//g' > pkg/sql/parser/gen/sql-gen.y.tmp || rm pkg/sql/parser/gen/sql-gen.y.tmp
Submodule 'c-deps/cryptopp' (https://github.com/cockroachdb/cryptopp.git) registered for path 'c-deps/cryptopp'
Submodule 'c-deps/googletest' (https://github.com/cockroachdb/googletest) registered for path 'c-deps/googletest'
Submodule 'c-deps/jemalloc' (https://github.com/cockroachdb/jemalloc.git) registered for path 'c-deps/jemalloc'
Submodule 'c-deps/krb5' (https://github.com/cockroachdb/krb5.git) registered for path 'c-deps/krb5'
Submodule 'c-deps/libedit' (https://github.com/cockroachdb/libedit.git) registered for path 'c-deps/libedit'
Submodule 'c-deps/protobuf' (https://github.com/cockroachdb/protobuf.git) registered for path 'c-deps/protobuf'
Submodule 'c-deps/rocksdb' (https://github.com/cockroachdb/rocksdb.git) registered for path 'c-deps/rocksdb'
Submodule 'c-deps/snappy' (https://github.com/cockroachdb/snappy.git) registered for path 'c-deps/snappy'
Submodule 'pkg/ui/yarn-vendor' (https://github.com/cockroachdb/yarn-vendored) registered for path 'pkg/ui/yarn-vendor'
Submodule 'vendor' (https://github.com/cockroachdb/vendored) registered for path 'vendor'
Cloning into '/go2/src/github.com/cockroachdb/cockroach/c-deps/cryptopp'...
mv -f pkg/sql/parser/gen/sql-gen.y.tmp pkg/sql/parser/gen/sql-gen.y
rm pkg/sql/parser/gen/types_regex.tmp
pkg/testutils/buildutil/build.go:20:2: cannot find package "github.com/pkg/errors" in any of:
	/go2/src/github.com/cockroachdb/cockroach/vendor/github.com/pkg/errors (vendor tree)
	/usr/local/go/src/github.com/pkg/errors (from $GOROOT)
	/go2/src/github.com/pkg/errors (from $GOPATH)
Makefile:1650: recipe for target 'bin/prereqs' failed
make: *** [bin/prereqs] Error 1
make: *** Waiting for unfinished jobs....
Cloning into '/go2/src/github.com/cockroachdb/cockroach/c-deps/googletest'...
Cloning into '/go2/src/github.com/cockroachdb/cockroach/c-deps/jemalloc'...
Cloning into '/go2/src/github.com/cockroachdb/cockroach/c-deps/krb5'...
Cloning into '/go2/src/github.com/cockroachdb/cockroach/c-deps/libedit'...
Cloning into '/go2/src/github.com/cockroachdb/cockroach/c-deps/protobuf'...
Cloning into '/go2/src/github.com/cockroachdb/cockroach/c-deps/rocksdb'...
Cloning into '/go2/src/github.com/cockroachdb/cockroach/c-deps/snappy'...
Cloning into '/go2/src/github.com/cockroachdb/cockroach/pkg/ui/yarn-vendor'...
Cloning into '/go2/src/github.com/cockroachdb/cockroach/vendor'...
...
```

--- a/Makefile
+++ b/Makefile
@@ -1646,7 +1646,7 @@ $(testbins): bin/%: bin/%.d | bin/prereqs $(SUBMODULES_TARGET)
 	mv -f $@.d.tmp $@.d
 	$(xgo) test $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LINKFLAGS)' -c -o $@ $($*-package)
 
-bin/prereqs: ./pkg/cmd/prereqs/*.go
+bin/prereqs: ./pkg/cmd/prereqs/*.go | bin/.submodules-initialized
 	@echo go install -v ./pkg/cmd/prereqs
 	@$(GO_INSTALL) -v ./pkg/cmd/prereqs
 

